### PR TITLE
Add hardware tests for timestamp

### PR DESCRIPTION
**Pull request by martiniil**
## Description

These changes add hardware tests to #223.

#223 is already approved & all hardware tests were performed.

### Things to add, update or check by the maintainers before this PR can be merged.

- [ ] ~Public api function documentation~
- [ ] ~Architecture documentation reflects actual code structure~
- [ ] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
- [ ] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
- [ ] ~Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))~
- [ ] ~Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))~
- [ ] ~CHANGELOG.rst updated~
- [x] Copyright headers
- [ ] ~Examples~

### Review Checklist
- [ ] ~Soft- and hardware architecture (diagrams and description)~
- [ ] Test review (test plan and individual test cases)
- [ ] Documentation describes purpose of file(s) and responsibilities
- [ ] Code (coding rules, style guide)

### Release planning (please answer)
- [ ] ~When is the new feature released?~
- [ ] ~Which dependent packages do have to be released simultaneously?~

### Hardware tests
- [x] Perform hardware tests

### CODE REVIEW

File: test/hw_tests/hwtest_timestamp.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+def is_increasing(values: list):
+    """ Check if the given list consists of monotonically increasing values.
+    """
+    return is_less_than(values[:-1], values[1:])
+
+
+class HwtestTimestamp(unittest.TestCase):
+    """ Check if the timestamp of the laserscan message
+        - monotonically increases,
+        - is bounded by the laserscan-msg receipt time.
```
Comment (ct2034) [inline test/hw_tests/hwtest_timestamp.py:43]: lower than the time the message was received

File: test/hw_tests/hwtest_timestamp.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def is_increasing(values: list):
+    """ Check if the given list consists of monotonically increasing values.
+    """
+    return is_less_than(values[:-1], values[1:])
+
+
+class HwtestTimestamp(unittest.TestCase):
+    """ Check if the timestamp of the laserscan message
```
Comment (ct2034) [inline test/hw_tests/hwtest_timestamp.py:41]: """ Check if the timestamp of the laserscan messages

File: test/hw_tests/hwtest_timestamp.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        rospy.Subscriber("/laser_1/scan", LaserScan, self.callback)
+
+    def callback(self, msg):
+        if len(self.received_stamps) < DESIRED_NUMBER_OF_STAMPS:
+            self.receipt_times.append(rospy.get_rostime())
+            self.received_stamps.append(msg.header.stamp)
+
+    def wait_for_msgs(self):
+        start_wait = rospy.Time.now()
+        while len(self.received_stamps) < DESIRED_NUMBER_OF_STAMPS:
```
Comment (ct2034) [inline test/hw_tests/hwtest_timestamp.py:58]: to make the node behave nicely on ctrl-c add: `and not rospy.is_shutdown()`

**Review by ct2034 — CHANGES_REQUESTED** [review]

File: standalone/test/acceptance_tests/acceptancetest_timestamps.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+```
+./acceptancetest_print_timestamps fragmented_scans true > log.txt
+```
+and with a smaller scan range (given in tenth of degree), for example:
+```
+./acceptancetest_print_timestamps angle_start -200 angle_end 200 > log.txt
+```
+
+## Comparison example
+The calculated timestamp of the laser-scan (first ray time) with scan counter `n` has to be greater than the timestamp of the last frame with scan counter `n-1` in wireshark and smaller than the timestamp of the first frame with scan counter `n` in wireshark. An example is shown in the screenshot below.
```
[the remark this answers is not in the document]
Comment (martiniil) [reply standalone/test/acceptance_tests/acceptancetest_timestamps.md:48]: It takes much effort. This is a first step.

**Review by ct2034 — APPROVED** [review]

**Pull request merged**